pipeline/metadata/flatten.py

```diff
@@ -366,13 +370,52 @@ def _process_satellite_v2(self, scan: Any,
         'error': scan.get('error', None),
         'anomaly': scan['anomaly'],
         'success': not scan['connect_error'],
+        'controls_failed': not scan['passed_control'],
         'received': None,
         'rcode': [],
         'measurement_id': random_measurement_id
     }
     received_ips = scan.get('response')
     yield from self._process_received_ips(row, received_ips)
 
+  def _process_satellite_v2_responses(
```

Comment:
Add a test for this function and also add an example responses_control.json file to the e2e test.

Reply:
Updated e2e test files and unit tests (`test_flattenmeasurement_dns`, `test_process_satellite_v2`)